web/modules/custom/adms_validator/src/AdmsValidationResult.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\adms_validator;
+
+use Drupal\rdf_entity\Database\Driver\sparql\Connection;
+use EasyRdf\Sparql\Result;
+
+/**
+ * A collection of Schema Errors.
+ */
+class AdmsValidationResult {
+
+  protected $errors = [];
+
+  /**
+   * Constructs a list of SchemaErrors from a query result.
+   *
+   * @param \EasyRdf\Sparql\Result $result
+   *   The result of the validation query.
+   * @param string $graph_uri
+   *   The graph URI.
+   * @param \Drupal\rdf_entity\Database\Driver\sparql\Connection $sparql
+   *   The SPARQL endpoint.
+   *
+   * @todo Remove $graph_uri, $sparql params in ISAICP-4296.
+   * @see https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-4296
+   */
+  public function __construct(Result $result, $graph_uri, Connection $sparql) {
+    foreach ($result as $error) {
+      // @todo Remove this hack in ISAICP-4296.
+      // @see https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-4296
+      if (!in_array($error->Rule_ID->getValue(), range(100, 103))) {
+        $this->errors[] = new SchemaError($error);
+      }
+    }
+    $this->applyRules100To103($graph_uri, $sparql);
+  }
+
+  /**
+   * Casts the schema errors to an array for rendering.
+   *
+   * @return array
+   *   Renderable data.
+   */
+  public function toRows() : array {
+    return array_map(function ($error) {
+      return (array) $error;
+    }, $this->errors);
+  }
+
+  /**
+   * The amount of errors in the list.
+   *
+   * @return int
+   *   Error count.
+   */
+  public function errorCount() : int {
+    return count($this->errors);
+  }
+
+  /**
+   * Returns TRUE if the validation is successful.
+   *
+   * @return bool
+   *   TRUE if the validation is successful.
+   */
+  public function isSuccessful(): bool {
+    return $this->errorCount() === 0;
+  }
+
+  /**
+   * Returns the errors in a table as a render array.
+   *
+   * @return array
+   *   A table render array containing the list of errors.
+   */
+  public function toTable(): array {
+    return [
+      '#theme' => 'table',
+      '#header' => [
+        t('Class name'),
+        t('Message'),
+        t('Object'),
+        t('Predicate'),
+        t('Rule description'),
+        t('Rule ID'),
+        t('Rule severity'),
+        t('Subject'),
+      ],
+      '#rows' => $this->toRows(),
+      '#empty' => t('No errors.'),
+    ];
+  }
+
+  /**
+   * Fix rules 100, 101, 102, 103.
+   *
+   * @param string $graph_uri
+   *   The graph URI.
+   * @param \Drupal\rdf_entity\Database\Driver\sparql\Connection $sparql
+   *   The SPARQL endpoint.
+   *
+   * @todo Remove this hack in ISAICP-4296.
+   * @see https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-4296
```

Comment:
I see the upstream issue included in this ticket is closed. What is the outcome of this?

Reply:
The upstream issue was opened based on a false assumption, so I closed that issue. Now, we moved this problem in ISAICP-4296 and this issue is no more blocked on ISAICP-4296 (because of this hack). So the QA of ISAICP-4296 and its upstream issue is not anymore in the scope of this QA.